lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,26 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
+      return if actions.empty? || actions.size < 1
+      grouped_actions = actions.group_by { |action, arg, source | LogStash::Outputs::ElasticSearch::VALID_HTTP_ACTIONS.include?(action) }
+      unless grouped_actions[false].nil?
+        @logger.warn("Number of requests filtered out before sending to Elasticsearch because of unsupported action, ", size: grouped_actions[false].size)
+        write_unsupported_events_to_dql(grouped_actions[false])
+      end
+      grouped_actions[true].nil? ? [] : grouped_actions[true]
+    end
+
+    def write_unsupported_events_to_dql(actions)
+      return if @dlq_writer.nil?
```

Comment:
`pipelines.yml` -> `dead_letter_queue.enable: true` didn't help. Is there any way I can make `common.rb` -> `dlq_enabled?` true?

Reply:
I would rely on `handle_dlq_status` implementation here, whose `action` argument is an `EventActionTuple` (an entry in the `actions` enumerator passed to `retrying_submit` that extends `Array` so it "splats" to an `action`,`args`,`source` but also has an `EventActionTuple#event` method to get the `LogStash::Event`)

From there, we can stub out the `handle_dlq_status` and validate it receives the args we expect.

Reply:
Well, I would agree to utilize `handle_dlq_status` if it _does not expect response information_. We are not sending any request to ES/3rd-party or cannot forcefully imitate the response (ex: with 400/0 code). It also logs the event and throws if DQL is not enabled where we are introducing log flood.

Reply:
We make a very serious effort in this plugin and others to not lose data, but when we need to drop data, at the very least we ensure that we are not silent about it. The log flood caused by routing through `handle_dlq_status ` is both intentional and necessary.